include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -478,12 +478,12 @@ using __repacked_tuple_t = typename __repacked_tuple<T>::type;
 template <typename _ContainerOrIterable>
 using __value_t = typename __internal::__memobj_traits<_ContainerOrIterable>::value_type;
 
-template <typename _T>
+template <typename _T, sycl::access_mode _AccessMode>
```

Comment:
Actually, the spec tells:
"4.7.6.9.2. Deduction tags for buffer command accessors
Some accessor constructors take a TagT parameter, which is used to deduce template arguments...."
And the implementation before (line 495) used that tag value - `sycl::read_write`.
`__acc(sycl::accessor(*__sycl_buf, __cgh, sycl::read_write, __dpl_sycl::__no_init{}))`

If we want to pass information about required access mode outside, lets accept the same tag as an additional argument in `__usm_or_buffer_accessor` constructor and pass one further into `accessor` constructor, like this: 
 ```
 template <typename TagT>
__usm_or_buffer_accessor(sycl::handler& __cgh, sycl::buffer<_T, 1>* __sycl_buf, TagT __tag)
        : __acc(sycl::accessor(*__sycl_buf, __tag, __dpl_sycl::__no_init{}))
```
That's all.

In that case we don't need a template parameter `sycl::access_mode _AccessMode` and don't  need `__get_access_mode_tag()` function.

Reply:
Or we can use another accessor constructor w/o tag parameter. 
`__acc(sycl::accessor(*__sycl_buf, __dpl_sycl::__no_init{}))`
An access mode for accessor is defined via accessor type definition, via `sycl::access_mode _AccessMode`

Reply:
Good point, fixed.